interview-questions.md

```diff
@@ -8,36 +8,103 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer: If you created a model cohort that has many_students and forgot to add the foreign key you can use rails g migration AddCohort cohort:references (or whatever you want to name the migration) which will add a cohort_id to the students database column. Once you've created the migration you use the command `rails db:migrate` the students table will have a new a new foreign key that references the id column of the cohort table.
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: The RESTful routes that need params to be passed are GET, PUT, DELETE, POST, PUT and delete. (HTTP verbs) and they must be passed params because they're used to update, modify, delete and add new data. Without params it wouldn't know how or what to change.
 
-Researched answer:
+Researched answer: I was correct in the RESTful routes that need params to be passed. After researching they are:
+GET which retrieves a resource by its ID.
+PUT or PATCH updates a resource by its given ID.
+DELETE deletes a resource by its given ID.
+POST creates a new resource. Which it needs params because it needs data in order to create a new resource.
+PUT or PATCH can also update multiple resources.
+DELETE /resource deletes ALL resources.
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: rails g model which creates a new file in the model folder
+rails g controller creates a new controller file in the controllers folder
+rails g migration creates a new database in the migrate folder.
 
 Researched answer:
+rails g model creates a new database model file in app/models director. It also creates a new database migration file in db/migrate directory. You would use this command by running rails g model Column_name name:string phone:integer
+____________
+rails g controller generates a new controller file in the controllers directory. to run this command you would use: rails g controller Whatever(your controller name) index(I'm just using index as an example) this command will create an index.html.erb view file as well as a pages_helper.rb helper file
+____________
+rails g migration generates a database migration file in the migrate folder. Migrations are used to add and modify database scheema. You would use rails g migration AddWhateverToModel name:string
+to update the database scheema you would then run rails db:migrate
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
 action: "GET" location: /students
+The controller method would be called would be GET/studens which the route would be index. The index performs a query to retrieve all of the student data from the database.
```

Comment:
You def have the right idea! Just flip them, and the controller method would be index.